Document Service#afterCommit handler [ECR-2728] (#252)
exonum/exonum-doc: src/get-started/java-binding.md

as an argument – a view that allows performing modifying operations. A service
schema object can be used to access data collections of this service.

Also, `Transaction#execute` method may throw `TransactionExecutionException`
which contains a transaction error report. This feature allows users to notify
Exonum about an error in a transaction execution whenever one occurs.

> It may check the preconditions before executing a transaction and either

Reviewer: It checks the preconditions...

Author: I'd keep it as is (or clarify), as a _transaction_ may (or may not) check the preconditions.

accepts it or throws an exception that is further transformed into an Exonum
core [TransactionResult enum][transaction-result] containing an error code and
a message with error data.
If transaction execution fails, the changes made by the transaction are
rolled back, while the error data is stored in the database for further user
reference. Light clients also provide access to information on the
[transaction][exonum-transaction] execution result
(which may be either success or failure) to their users.

> ### Blockchain events

Reviewer: Events

> A service can also handle a block commit event, that occurs each time

Reviewer: delete comma

[transaction][exonum-transaction] execution result
(which may be either success or failure) to their users.

### Blockchain Events

> A service can also handle a block commit event that occurs each time

Reviewer 2: ```suggestion
A service may handle a block commit event that occurs each time
```

Author: I'd keep can, as it seems to fit here better (describing a capability)

the framework commits a new block. The framework delivers this event to

> implementations of [`Service#afterCommit(BlockCommittedEvent)`][service-after-commit]

Reviewer 2: such as "callback" word is used below:
"implementations of the callback method..."

in each deployed service. Each node in the network processes that event
independently from other nodes. The event includes a `Snapshot`,
allowing a read-only access to the database state _exactly_ after the commit
of the corresponding block.

As services can read the database state in the handler, they may detect
any changes in it, e.g., that a certain transaction is executed;
or some condition is met. Services may also create and submit new transactions

> using [`Node#submitTransaction`][node-submittransaction]. Using this callback

Reviewer 3: Nit: why do we use `submittransaction` here, but `after-commit` above? (I like hyphens better)

- Follow the instructions in the [application guide][app-tutorial] to configure
  and start an Exonum node with your service.

## Common Library

Java Binding includes a library module that can be useful for Java client
applications that interact with an Exonum service and

> does not have the dependency on Java Binding Core. The module contains Java

Reviewer: Java Binding includes a library module that can be useful for Java client
applications that interact with an Exonum service. The module does not have the dependency on Java Binding Core, but it contains Java classes obligatory for core that can now as well be easily applied in clients, if necessary.

## Common Library

Java Binding includes a library module that can be useful for Java client
applications that interact with an Exonum service. The module does not
have the dependency on Java Binding Core, but it contains Java classes

> obligatory for core that can now as well be easily used in clients,

Reviewer 2: ```suggestion
obligatory for the core that can now as well be easily used in clients,
```
